modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/application/mgt/ServiceProviderUserRoleValidationTestCase.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    private void createApplications() throws Exception {
+
+        log.info("Creating the service providers required for the testcase.");
+        for (int i = 1; i < 10; i++) {
+            ServiceProvider serviceProvider = new ServiceProvider();
+            serviceProvider.setApplicationName(APPLICATION_PREFIX + i);
+            serviceProvider.setDescription(SERVICE_PROVIDER_DESCRIPTION);
+            applicationManagementServiceClient.createApplication(serviceProvider);
+        }
+    }
+
+    private void deleteApplications() throws Exception {
+
+        log.info("Deleting the service providers used during the testcase.");
+        for (int i = 1; i < 10; i++) {
+            applicationManagementServiceClient.deleteApplication(APPLICATION_PREFIX + i);
+        }
+    }
+
+    private void createApplicationRoles() throws Exception {
+
+        log.info("Creating the application roles required for the testcase.");
+        for (int i = 1; i <= 5; i++) {
+            remoteUSMServiceClient.addRole(ROLE_PREFIX + i, new String[0], null);
+        }
+    }
+
+    private void deleteApplicationRoles() throws Exception {
+
+        log.info("Deleting the application roles used during the testcase.");
+        for (int i = 1; i <= 5; i++) {
+            remoteUSMServiceClient.deleteRole(ROLE_PREFIX + i);
+        }
+    }
+
+    private void updateAdminUser() throws Exception {
+
+        log.info("Updating the role list of the user : " + USERNAME + ".");
+        remoteUSMServiceClient.updateRoleListOfUser(USERNAME, null, ROLES);
+    }
+
+    @Test(description = "Retrieve application count for the admin user without filter.")
+    public void testApplicationCountForUserWithoutFilter() throws Exception {
+
+        int count = applicationManagementServiceClient.getCountOfAllApplications();
+        Assert.assertEquals(count, APPLICATION_COUNT_WITHOUT_FILTER,
+                String.format("The expected application count without a filter does not match the actual application " +
+                                "count: Expected Application Count: %d Actual Application Count: %d.",
+                        APPLICATION_COUNT_WITHOUT_FILTER, count)
+        );
+    }
+
+    @Test(description = "Retrieve application count for the admin user with filter.")
+    public void testApplicationCountForUserWithFilter() throws Exception {
+
+        int count = applicationManagementServiceClient.getCountOfApplications(APPLICATION_FILTER);
+        Assert.assertEquals(count, APPLICATION_COUNT_WITH_FILTER,
+                String.format("The expected application count with a filter does not match the actual application " +
+                                "count: Expected Application Count: %d Actual Application Count: %d.",
+                        APPLICATION_COUNT_WITH_FILTER, count)
+        );
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void clearObjects() throws Exception {
+
+        deleteObjects();
+        clear();
```

Comment:
In after class method, we should restore to the default configuration without restart. Refer this https://github.com/wso2/product-is/blob/0688b361ec4ad514a67ed6848e2b2e15cf6c8a26/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/provisioning/DBSeperationTestCase.java#L71

Reply:
Fixed with https://github.com/wso2/product-is/pull/13308/commits/1d02ae0e5259cfa0048206403bf8503c61b05387.